lib/skate_web/channels/vehicles_search_channel.ex

```diff
@@ -0,0 +1,76 @@
+defmodule SkateWeb.VehiclesSearchChannel do
+  use SkateWeb, :channel
+  use SkateWeb.AuthenticatedChannel
+  require Logger
+
+  alias Realtime.Server
+  alias Util.Duration
+
+  @impl SkateWeb.AuthenticatedChannel
+  def join_authenticated(
+        "vehicles_search:",
+        %{"limit" => limit, "text" => text, "property" => property},
+        socket
+      ) do
+    username_from_socket! =
+      Application.get_env(
+        :skate,
+        :username_from_socket!,
+        &SkateWeb.AuthManager.username_from_socket!/1
+      )
+
+    username = username_from_socket!.(socket)
+    %{id: user_id} = Guardian.Phoenix.Socket.current_resource(socket)
+
+    subscribe_args = %{
+      property: String.to_existing_atom(property),
+      limit: limit,
+      text: text
+    }
+
+    subscribe_args =
+      Map.put(
+        subscribe_args,
+        :include_logged_out_vehicles,
+        Skate.Settings.User.is_in_test_group(user_id, "search-logged-out-vehicles")
+      )
+
+    Logger.info(fn ->
+      "User=#{username} searched for property=#{subscribe_args.property}, text=#{subscribe_args.text} limit=#{subscribe_args.limit}"
+    end)
+
+    vehicles = Duration.log_duration(Server, :subscribe_to_search, [subscribe_args])
+
+    {:ok, %{data: vehicles}, socket}
+  end
+
+  @impl Phoenix.Channel
+  def handle_in(
+        "update_search_query",
+        %{"text" => text, "limit" => limit, "property" => property},
+        socket
+      ) do
+    vehicles =
+      Duration.log_duration(Server, :update_search_subscription, [
```

Comment:
Thanks for raising this! I hadn't considered the effects of an update of vehicle data between these two calls. 

>  Since this is calling Server.update_search_description directly and that function along with the private update_subscription/2 does all of its logic without making a GenServer call, this means that all of the calls to the registry functions happen in the calling process, not the Realtime.Server process. 

I _think_ users would still get the latest data for the newest search because the call to `lookup({ets, subscription_key})` from within `update_subscription` would return the latest data available in the `ets` table, even if that `ets` table was updated between the `unregister_match` and `register` calls. But there may still be some nuance about the timing & GenServer that I'm missing.

Reply:
I'm referring to what happens if the Concentrate GenServer pushes updated vehicle information to the `Realtime.Server` process. In that case `Server.broadcast/2` would be called and it would use the subscription information available from Registry at that time. That said, I do still believe it's an extreme edge case so I'm not going to worry too much about it.